packages/lesswrong/server/repos/PostRelationsRepo.ts

```diff
@@ -0,0 +1,25 @@
+import abstractRepo from "./AbstractRepo";
+
+@abstractRepo.register()
+class PostRelationsRepo extends abstractRepo<PostRelationsRepo>() {
+  getPostRelationsByPostId(postId: string, depth = 3): Promise<DbPostRelation[]> {
+    return this.db.any(`
+      WITH RECURSIVE search_tree(
```

Comment:
That's a good point. I feel like it still makes sense for the default to be a Typescript string though since most will probably be pretty short, and also because it may sometimes be desirable/necessary to use template interpolation.

Note that `pg-promise` also has this: [https://vitaly-t.github.io/pg-promise/QueryFile.html](https://vitaly-t.github.io/pg-promise/QueryFile.html). We could probably just put some `executeFile` method on AbstractRepo that handles using that correctly (that's "correctly" in the sense that you've got to be careful to cache the `QueryFile` objects to avoid unneccesary duplicated file reads).

Reply:
Please don't. Putting them in `.sql` files would make my life much worse.